### DISCUSSION TITLE
btConvex2dShape debug drawing fix (2.87 stable)

### DISCUSSION THREAD

**Discussion by ruby0x1** [bulletphysics/bullet3#3584] [General]
The convex2d shapes don't draw their child shape as expected in 2.87 stable.

Inside btCollisionWorld.cpp in the `debugDrawObject` function, the `COMPOUND_SHAPE_PROXYTYPE` is handling it's children, so I added an else which draws them for the `CONVEX_2D_SHAPE_PROXYTYPE` as well.

```c++
} else if (shape->getShapeType() == CONVEX_2D_SHAPE_PROXYTYPE)
{
    const btConvex2dShape* convex2dShape = static_cast<const btConvex2dShape*>(shape);
    const btCollisionShape* childShape = convex2dShape->getChildShape();
    debugDrawObject(worldTransform,childShape,color);
} else
```

After fixing it, wireframes correctly show up for these shapes. 
I would PR but I'm pressed for time and file an issue to remind myself. 
Feel free to fix it if this is the correct/intended approach!

![screen shot 2018-09-06 at 7 54 36 pm](https://user-images.githubusercontent.com/236374/45196184-e15ca980-b20f-11e8-9f81-c947f3677d9a.png)

**Comment by bilenkonito** [comment]
This is still a problem in 2.88. Can we please get an official response?
